packages/validator/src/services/attestationDuties.ts

```diff
@@ -184,10 +190,15 @@ export class AttestationDutiesService {
 
     // If there are any subscriptions, push them out to the beacon node.
     if (beaconCommitteeSubscriptions.length > 0) {
-      // TODO: Should log or throw?
-      await this.api.validator.prepareBeaconCommitteeSubnet(beaconCommitteeSubscriptions).catch((e: Error) => {
-        throw extendError(e, "Failed to subscribe to beacon committee subnets");
-      });
+      const subscriptionsBatches = batchItems(beaconCommitteeSubscriptions, {batchSize: SUBSCRIPTIONS_PER_REQUEST});
+      await Promise.all(
```

Comment:
do we want to loop and await each chunk rather than parallel throw?

Reply:
@tuyennhv I would like to compute the real max size of this call. The limiting factor should be request size.

Reply:
@g11tech I think parallel await is better here since we don't need to wait for each request to fully resolve before triggering the next.

Reply:
@dapplion a request size is around 107 bytes, make it 120 to buffer, typically a server accept a max body size of 1MB. I changed the constant to `Math.floor(2 ** 20 / 120) = 8738`

so for the 10k validators node, it takes only 2 requests. Tested in that node and it works well.

I added all of this in the comment of `SUBSCRIPTIONS_PER_REQUEST` constant

Reply:
can we atleast wrapError these promises and throw a single error (incase that happens) else it would fill up the screen in case there are 3+ such requests, not sure how we can achieve something similar on the bn side

Reply:
another reason regarding sending this one by one could be: that bn will process requests one by one even if we send them in parallel, and there could be timeout issues for lining up the calls, though don't know what would be request to response timeline for a full chunk

Reply:
@g11tech `Promise.all()` will reject immediately upon any promise reject https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all